packages/medusa/src/api/routes/admin/inventory-items/delete-location-level.ts

```diff
@@ -0,0 +1,81 @@
+import { Request, Response } from "express"
+import { MedusaError } from "medusa-core-utils"
+import { IInventoryService } from "../../../../interfaces"
+
+/**
+ * @oas [delete] /inventory-items/{id}/location-levels/{location_id}
+ * operationId: "DeleteInventoryItemsInventoryIteLocationLevelsLocation"
+ * summary: "Delete a location level of an Inventory Item."
+ * description: "Delete a location level of an Inventory Item."
+ * x-authenticated: true
+ * parameters:
+ *   - (path) id=* {string} The ID of the Inventory Item.
+ *   - (path) location_id=* {string} The ID of the location.
+ * x-codeSamples:
+ *   - lang: JavaScript
+ *     label: JS Client
+ *     source: |
+ *       import Medusa from "@medusajs/medusa-js"
+ *       const medusa = new Medusa({ baseUrl: MEDUSA_BACKEND_URL, maxRetries: 3 })
+ *       // must be previously logged in or use api token
+ *       medusa.admin.inventoryItems.deleteLocationLevel(inventoryItemId, locationId)
+ *       .then(({ inventory_item }) => {
+ *         console.log(inventory_item.id);
+ *       });
+ *   - lang: Shell
+ *     label: cURL
+ *     source: |
+ *       curl --location --request DELETE 'https://medusa-url.com/admin/inventory-items/{id}/location-levels/{location_id}' \
+ *       --header 'Authorization: Bearer {api_token}' \
+ *       --header 'Content-Type: application/json'
+ * security:
+ *   - api_token: []
+ *   - cookie_auth: []
+ * tags:
+ *   - Inventory Items
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           $ref: "#/components/schemas/AdminInventoryItemsRes"
+ *   "400":
+ *     $ref: "#/components/responses/400_error"
+ *   "401":
+ *     $ref: "#/components/responses/unauthorized"
+ *   "404":
+ *     $ref: "#/components/responses/not_found_error"
+ *   "409":
+ *     $ref: "#/components/responses/invalid_state_error"
+ *   "422":
+ *     $ref: "#/components/responses/invalid_request_error"
+ *   "500":
+ *     $ref: "#/components/responses/500_error"
+ */
+export default async (req: Request, res: Response) => {
+  const { id, location_id } = req.params
+
+  const inventoryService: IInventoryService =
+    req.scope.resolve("inventoryService")
+
+  const reservedQuantity = await inventoryService.retrieveReservedQuantity(id, [
+    location_id,
+  ])
+
+  if (reservedQuantity > 0) {
+    throw new MedusaError(
+      MedusaError.Types.NOT_ALLOWED,
+      `Cannot remove Inventory Level ${id} at Location ${location_id} because there are reserved items.`
+    )
+  }
+
+  await inventoryService.deleteInventoryLevel(id, location_id)
```

Comment:
todo: If I am not mistaken, the inventory service is a singleton that extends the TransactionBaseService. If some places call the service with a transaction, the concurrent requests will use the same instance and therefore the same transaction. Once the transaction will be committed or rolled back, we will receive an error already release. 

Correct me if I am missing something. otherwise, we have to go through the different end points and fix that.